src/peft/tuners/tuners_utils.py

```diff
@@ -359,8 +359,8 @@ def check_target_module_exists(config, key: str) -> bool | re.Match[str] | None:
     if isinstance(config.target_modules, str):
         target_module_found = re.fullmatch(config.target_modules, key)
     else:
-        target_module_found = any(re.match(f".*\.{target_key}$", key) for target_key in config.target_modules) or any(
-            target_key == key for target_key in config.target_modules
+        target_module_found = key in config.target_modules or any(
+            key.endswith(f".{target_key}") for target_key in config.target_modules
```

Comment:
Could you please explain why this change was made? Is it related to the other changes in this PR or just to simplify?